### PR TITLE
Implement methods #1

### PR DESCRIPTION
**Pull request by Fahim-Khan-P**
### I added the following:**

- [ ] Create a Book class in a separate .rb file.
- [ ] Create a `Label` class with an association to the `Item `class (in a separate .rb file).
- [ ] All `Book `class properties visible in the diagram should be defined and set up in the constructor method.
- [ ] All `Label `class properties visible in the diagram should be defined and set up in the constructor method.
- [ ] Implement methods:
`add_item` method in the `Label `class
should take an instance of the `Item `class as an input
should add the input item to the collection of items
should add self as a property of the item object (by using the correct setter from the item object)
`can_be_archived?()` in the `Book `class
should override the method from the parent class
should return **true** if parent's method returns **true** OR if **_cover_state equals to "bad"_**
otherwise, it should return **false**

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(publisher, cover_state, publish_date)
+    super(publish_date)
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  def can_be_archived?
+    super || @cover_state == 'bad'
+  end
+end
```
Comment (tonnymuchui) [inline book.rb:15]: Book Class Added well :heavy_check_mark:

File: item.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def add_genre(new_genre)
+    @genre = new_genre
+    genre.add_item(new_genre)
+  end
+
+  def add_author(new_author)
+    @author = new_author
+    author.add_item(new_author)
+  end
```
Comment (tonnymuchui) [inline item.rb:23]: Item methods added successfully :heavy_check_mark:

**Review by tonnymuchui — APPROVED** [review]
## Status: Approved ✔️
Hi Micronaut @Fahim-Khan-P 
Great work!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Route Works✔️
- Implement classes :heavy_check_mark: 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @tonnymuchui  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Fahim-Khan-P** [comment]
Thanks @tonnymuchui

**Pull request merged**